packages/rtk-query-graphql-request-base-query/src/index.ts

```diff
@@ -3,17 +3,18 @@ import type { BaseQueryFn } from '@reduxjs/toolkit/query'
 import type { DocumentNode } from 'graphql'
 import { GraphQLClient, ClientError } from 'graphql-request'
 import type {
+  ErrorResponse,
   GraphqlRequestBaseQueryArgs,
   PrepareHeaders,
   RequestHeaders,
 } from './GraphqlBaseQueryTypes'
 
-export const graphqlRequestBaseQuery = (
-  options: GraphqlRequestBaseQueryArgs
+export const graphqlRequestBaseQuery = <T = ErrorResponse>(
+  options: GraphqlRequestBaseQueryArgs<T>
 ): BaseQueryFn<
   { document: string | DocumentNode; variables?: any },
   unknown,
-  Pick<ClientError, 'name' | 'message' | 'stack'>,
+  Pick<ClientError, 'name' | 'message' | 'stack'> | T,
```

Comment:
```suggestion
  E,
```

Reply:
By the logical point of view shoud works, but i'm tryng and unfortunately typization doesn't works. 
This because  typescript infer `customizedError` to
```
E | {
    name: string;
    message: string;
    stack: string;
}
```
But using generics initialization our type is just `E` that can be `ErrorResponse`


To make it works I added this (is more or less the part you drop)

```suggestion
  ErrorResponse | E,
```

Reply:
The build fails on pipeline, I need to do some other check.

Reply:
In the last version i apply your suggestions, but i need to force `customizedErrors` to `E`. 
I think the main problem is that `E` can be initialized to `unknown` by `BaseQueryFn`  and this cause some conflict.